.github/workflows/linux_unit_tests.yml

```diff
@@ -0,0 +1,79 @@
+name: Linux Unit Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+  push:
+    branches:
+      - main
+
+jobs:
+  unit_tests:
+    name: ${{ matrix.python_version }} unit tests, Core Dependencies (${{ matrix.core_dependencies }}), Codecov (${{ matrix.codecov }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        include:
+          - python_version: "3.7"
+            core_dependencies: false
+            codecov: false
+          - python_version: "3.8"
+            core_dependencies: false
+            codecov: false
```

Comment:
@ParthivNaresh this should be `true`, right? Our old CI collects `coverage` report data on python 3.8 both with and without `core_dependencies`. That way, we cover `if` statements which are only checked in one of the two cases, and then `codecov.io` merges the two coverage reports. Does that make sense?

Reply:
@gsheni does the fact that codecov.io merges these reports make it better? Just want to make sure we're all on the same page haha

Reply:
Oh I wasn't aware that CodeCov merges report. yes, that is fine, and @dsherry is correct
- https://docs.codecov.io/docs/merging-reports

Reply:
Yeah our CI doesn't make it super obvious that that's what's going on under the hood, lol. Because we just ship both the `core_deps=false` and `core_deps=true` runs over to `codecov.io` and then they handle merging (summing) the reports. But yep, we need both at the moment!